routers/api/v1/repo/file.go

```diff
@@ -294,7 +295,53 @@ func GetArchive(ctx *context.APIContext) {
 		defer gitRepo.Close()
 	}
 
-	repo.Download(ctx.Context)
+	archiveDownload(ctx)
+}
+
+func archiveDownload(ctx *context.APIContext) {
```

Comment:
Note: this is _very_ similar to `web/repo/repo.go#Download` (only difference is the type of context). That is unfortunate but I didn't see any obvious way of using the same code for these since they produce very different error responses (json vs html).